### PR TITLE
Stat: Cut the top of the chart issue

### PR DESCRIPTION
**Pull request by Develer**
**What is this feature?**

This PR fixes the issue when chart data is cut top and doesn't adjust to the new data chart size.

Before:

https://github.com/grafana/grafana/assets/1418924/7afce500-0394-4ce8-8792-1357ccd8d571

After:

https://github.com/grafana/grafana/assets/1418924/76466439-d3f4-483b-9e0d-15ba6286e1c6

**Which issue(s) does this PR fix?**:

Fixes [#10667-stat-panel-chart-](https://github.com/grafana/support-escalations/issues/10667)

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/uPlot/Plot.tsx [hunk 1]
```diff
@@ -83,12 +83,7 @@ export class UPlotChart extends Component<PlotProps, UPlotChartState> {
...
     let { plot } = this.state;
 
-    if (!sameDims(prevProps, this.props)) {
-      plot?.setSize({
-        width: Math.floor(this.props.width),
-        height: Math.floor(this.props.height),
-      });
-    } else if (!sameConfig(prevProps, this.props)) {
+    if (!sameDims(prevProps, this.props) || !sameConfig(prevProps, this.props)) {
       this.reinitPlot();
```
Comment (drew08t) [inline packages/grafana-ui/src/components/uPlot/Plot.tsx:87]: Does reinitPlot() not also calculate the size?

Comment (drew08t) [reply packages/grafana-ui/src/components/uPlot/Plot.tsx:87]: More specifically, @leeoniya, is there a reason for optimization to separate these two conditions?

Comment (leeoniya) [reply packages/grafana-ui/src/components/uPlot/Plot.tsx:87]: yes. you dont want to reinit the plot on every pixel of a panel resize!

the solution should be to make sure that the stat panel re-inits the config when the min or max change.

File: packages/grafana-ui/src/components/uPlot/Plot.tsx [hunk 2]
```diff
@@ -83,12 +83,7 @@ export class UPlotChart extends Component<PlotProps, UPlotChartState> {
   componentDidUpdate(prevProps: PlotProps) {
     let { plot } = this.state;
 
-    if (!sameDims(prevProps, this.props)) {
-      plot?.setSize({
```
Comment (Develer) [inline packages/grafana-ui/src/components/uPlot/Plot.tsx:87]: @leeoniya shouldn't `plot.setSize` do the job? Looks like the config doesn't change, at least explicitly.

Comment (Develer) [reply packages/grafana-ui/src/components/uPlot/Plot.tsx:87]: I think I found the problem, we are checking data structure but not data itself. That's why is doesn't trigger config update.
https://github.com/grafana/grafana/blob/d4c28aa7574e32b8c5e01efd255b685096c2378f/packages/grafana-ui/src/components/Sparkline/Sparkline.tsx#L88-L90

**Review by leeoniya — APPROVED** [review]

**Review by drew08t — APPROVED** [review]
Looks good!

**Pull request merged**